controllers/deployment.go

```diff
@@ -227,7 +228,7 @@ func buildStatefulSets(master *v1alpha1.CDAPMaster, name string, services Servic
 }
 
 // Return a single single-/multi- container deployment containing a list of supplied services
-func buildDeployment(master *v1alpha1.CDAPMaster, name string, services ServiceGroup, labels map[string]string, cconf, hconf, dataDir string) (*DeploymentSpec, error) {
+func buildDeployment(master *v1alpha1.CDAPMaster, name string, services ServiceGroup, labels map[string]string, cconf, hconf, sysappconf, dataDir string) (*DeploymentSpec, error) {
```

Comment:
The list of params is getting too long here. Let's refactor by creating a config class that encapsulates cconf, hconf and sysconf and pass that a single param. Either make the refactoring in this change or a follow-up one.

Reply:
Will take care of it in future refactoring.